libraries/ms-common/inc/can_hw.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+// CAN HW Interface
+//
+// Used to initiate CAN TX and RX directly through the MCU, without any preprocessing or queues.
+// Note that none of our systems currently support more than one CAN interface natively.
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include "can_hw_mcu.h"
+#include "status.h"
+
+// Used to process HW events within the CAN ISR, ideally as short as possible.
+typedef void (*CANHwEventHandlerCb)(void *context);
+
+typedef struct CANHwEventHandler {
+  CANHwEventHandlerCb callback;
+  void *context;
+} CANHwEventHandler;
+
+typedef enum {
+  CAN_HW_EVENT_TX_READY = 0,
+  CAN_HW_EVENT_MSG_RX,
+  CAN_HW_EVENT_BUS_ERROR,
+  NUM_CAN_HW_EVENTS
+} CANHwEvent;
+
+typedef enum {
+  CAN_HW_BUS_STATUS_OK = 0,
+  CAN_HW_BUS_STATUS_ERROR,
+  CAN_HW_BUS_STATUS_OFF
+} CANHwBusStatus;
+
+typedef struct CANHwConfig {
+  CANHwBase base;
+  uint16_t bus_speed; // in kbps
+  uint8_t num_filters;
+  bool loopback;
+  CANHwEventHandler handlers[NUM_CAN_HW_EVENTS];
+} CANHwConfig;
```

Comment:
The other drivers follow a different paradigm (uses an enum to index into hidden storage) while the CAN drivers are more of a catch-all storage/settings struct.

This is mostly because I wrote these parts last year before all of this existed. I'm not entirely sure which approach is better. Thoughts?